netmiko/base_connection.py

```diff
@@ -617,12 +617,16 @@ def _open_ssh(self, ssh_channel_class: Optional[Type[SSHChannel]]) -> None:
             ChannelClass = SSHChannel
         else:
             ChannelClass = ssh_channel_class
+        assert issubclass(ChannelClass, SSHChannel)
```

Comment:
I *BELIEVE* this is safe and correct.  if it's not would need to either make it true, or do something else to make sure it's okay to provide `use_keys` and `password` parameters.